src/Provider/ArcGISOnline/ArcGISOnline.php

```diff
@@ -45,14 +50,45 @@ final class ArcGISOnline extends AbstractHttpProvider implements Provider
     private $sourceCountry;
 
     /**
+     * @var string
+     *
+     * Currently valid ArcGIS World Geocoding Service token.
+     * https://developers.arcgis.com/rest/geocode/api-reference/geocoding-authenticate-a-request.htm
+     */
+    private $token;
+
+    /**
+     * ArcGIS World Geocoding Service.
+     * https://developers.arcgis.com/rest/geocode/api-reference/overview-world-geocoding-service.htm.
+     *
      * @param HttpClient $client        An HTTP adapter
+     * @param string     $token         Your authentication token
      * @param string     $sourceCountry Country biasing (optional)
+     *
+     * @return GoogleMaps
```

Comment:
Why GoogleMaps ?

Reply:
Ah, that's a remnant; I used the business function from the Google Maps provider to structure this function. I'll update it.